Sources/BraintreeCore/BTHTTP.m

```diff
@@ -295,6 +338,12 @@ - (void)handleRequestCompletion:(NSData *)data response:(NSURLResponse *)respons
         }
         return;
     }
+    
+    // We should only cache the response if we do not have an error
+    if (request != nil && shouldCache) {
```

Comment:
We might need to check the HTTP response code. I'm not sure all the possible ones the gateway can produce, but  anything in the `2xx` range is probably good to cache.

Reply:
Oh ok I didn't see the checks in the diff I had to expand the code fold 🙌